pkg/blockbuilder/tsdb.go

```diff
@@ -133,12 +133,12 @@ func (b *TSDBBuilder) Process(ctx context.Context, rec *kgo.Record, lastBlockMax
 		ref, copiedLabels := app.GetRef(nonCopiedLabels, hash)
 
 		for _, s := range ts.Samples {
-			if s.TimestampMs >= blockMax {
+			if !processEverything && s.TimestampMs >= blockMax {
 				// We will process this sample in the next cycle.
 				allSamplesProcessed = false
 				continue
 			}
-			if recordAlreadyProcessed && s.TimestampMs < lastBlockMax {
+			if !processEverything && recordAlreadyProcessed && s.TimestampMs < lastBlockMax {
```

Comment:
As I understand it the first time we restart blockbuilders with this flag enabled, we will produce a small amount of duplicate records.  (records that had `recordAlreadyProcessed` from the previous cycle, but we're now ignoring those.) I do not see an avenue where we'd _skip_ any records on the first run. Am I right?

Reply:
Yes, migrating from old the new way will create some duplicate samples in blocks which is harmless since compactor will merge them together.